src/mainwindow.cpp

```diff
@@ -5133,9 +5133,7 @@ void MainWindow::on_actionSettings_triggered()
       ui->dataTabShowFromArchives->setCheckState(Qt::Checked);
       ui->dataTabShowFromArchives->setEnabled(true);
     }
-    m_OrganizerCore.refreshModList();
-    m_OrganizerCore.refreshDirectoryStructure();
-    m_OrganizerCore.refreshLists();
+    m_OrganizerCore.refresh();
   }
```

Comment:
Uh I guess we got some optimization here, nice one. Hopefully it wasn't needed for some convoluted reason.

Reply:
The call to `refreshDirectoryStructure` was likely doing nothing (unless the directory update from `refreshModList` had already finished... ) due to the `m_DirectoryUpdate` "protection". Even with a very small setup, I doubt the update would have finished when reaching this line.

The `refreshLists` was doing something, but with the outdated directory structure, and this was override by the call to `refreshLists` in the slot at the end of the directory structure refresh. To me this looks like a left-over from a previous implementation where `refreshModList()` would  not refresh the directory structure or to a mistake due to the name `refreshModList()`.